src/fund/policies/risk-management/MaxPositions.sol

```diff
@@ -1,35 +1,48 @@
 // SPDX-License-Identifier: GPL-3.0
 pragma solidity 0.6.8;
 
-import "../../hub/Spoke.sol";
-import "../../shares/Shares.sol";
 import "../../vault/Vault.sol";
-import "../TradingSignatures.sol";
+import "../utils/CallOnIntegrationPostValidatePolicyBase.sol";
 
 /// @title MaxPositions Contract
 /// @author Melon Council DAO <security@meloncoucil.io>
 /// @notice Validates the allowed number of owned assets of a particular fund
-contract MaxPositions is TradingSignatures {
-    enum Applied { pre, post }
+contract MaxPositions is CallOnIntegrationPostValidatePolicyBase {
+    event MaxPositionsSet(address policyManager, uint256 value);
 
-    uint public maxPositions;
+    mapping (address => uint256) public policyManagerToMaxPositions;
 
-    /// @dev _maxPositions = 10 means max 10 different asset tokens
-    /// @dev _maxPositions = 0 means no asset tokens are investable
-    constructor(uint _maxPositions) public { maxPositions = _maxPositions; }
+    constructor(address _registry) public PolicyBase(_registry) {}
 
-    function rule(bytes4 sig, address[5] calldata addresses, uint[3] calldata values, bytes32 identifier)
+    // EXTERNAL FUNCTIONS
+
+    /// @notice Add the initial policy settings for a fund
+    /// @param _encodedSettings Encoded settings to apply to a fund
+    /// @dev A fund's PolicyManager is always the sender
+    /// @dev Only called once, on PolicyManager.enablePolicies()
+    function addFundSettings(bytes calldata _encodedSettings) external override onlyPolicyManager {
+        uint256 maxPositions = abi.decode(_encodedSettings, (uint256));
+        require(maxPositions > 1, "addFundSettings: maxPositions must be greater than 1");
```

Comment:
I don't think so, because the denomination asset is included, so the fund would be unusable (investors send asset1, and the fund manager can't make any trades/loans/etc for an asset2)